qcodes/dataset/data_set_protocol.py

```diff
@@ -41,13 +40,15 @@
     import pandas as pd
     import xarray as xr
 
+    from qcodes.instrument.parameter import _BaseParameter
+
     from .data_set_cache import DataSetCache
 
 array_like_types = (tuple, list, np.ndarray)
 scalar_res_types = Union[str, complex, np.integer, np.floating, np.complexfloating]
 values_type = Union[scalar_res_types, np.ndarray, Sequence[scalar_res_types]]
-res_type = Tuple[Union[_BaseParameter, str], values_type]
-setpoints_type = Sequence[Union[str, _BaseParameter]]
+res_type = Tuple[Union["_BaseParameter", str], values_type]
+setpoints_type = Sequence[Union[str, "_BaseParameter"]]
```

Comment:
in this module stringifying is not needed due to `from __future__ import annotations` , right?

Reply:
Unfortunatly the string repr is still needed in this context https://www.python.org/dev/peps/pep-0563/#forward-references